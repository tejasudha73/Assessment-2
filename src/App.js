import React, { Component } from 'react';
import { Button, Grid, Row, Col, Panel, Checkbox, FormGroup, FormControl, ControlLabel } from 'react-bootstrap';

const adultsOptions = [{value: '1',text: '1' }, {value: '2',text: '2' }];
const childrenOptions = [{value: '0',text: '0' }, {value: '1',text: '1' }, {value: '2',text: '2' }];
const adults = adultsOptions.map(adult => {
  return (
    <option value={adult.value} key={adult.value}>{adult.text}</option>
  );
});
const children = childrenOptions.map(child => {
  return (
    <option value={child.value} key={child.value}>{child.text}</option>
  );
});

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      rooms : [
                {key: 1, text:'Room 1', checked: true,  adult: adultsOptions[0].value, child: childrenOptions[0].value},
                {key: 2, text:'Room 2', checked: false, adult: adultsOptions[0].value, child: childrenOptions[0].value},
                {key: 3, text:'Room 3', checked: false, adult: adultsOptions[0].value, child: childrenOptions[0].value},
                {key: 4, text:'Room 4', checked: false, adult: adultsOptions[0].value, child: childrenOptions[0].value}
              ],
      adult: adultsOptions[0].value,
      child: childrenOptions[0].value,
      checked: false
    };
    this.onRoomSelect = this.onRoomSelect.bind(this);
    this.submit = this.submit.bind(this);
    this.setChild = this.setChild.bind(this);
    this.setAdult = this.setAdult.bind(this);
  }
  componentDidMount() {
    let allSelectedRooms = JSON.parse(localStorage.getItem('rooms')) || this.state.rooms;
    this.setState({rooms: allSelectedRooms});
  }
  onRoomSelect(event, data) {
    event.persist();
    this.state.rooms.filter((room) => {
      if (room.checked) {
        room.checked = room.key === data.key || room.key >= data.key ? !room.checked : room.checked;
      } else {
        room.checked = room.key === data.key || room.key <= data.key ? !room.checked : room.checked;
      }
    });
    this.setState({checked: event.target.value === 'on'});
  }
  submit () {
    setDataToLocalstorage(this.state.rooms);
  }
  setAdult(event, data) {
    event.persist();
    this.state.rooms.filter((room) => {
      if (room.key === data.key) {
        data.adult = event.target.value;
      }
    });
    this.setState({adult: event.target.value});
    // setDataToLocalstorage(this.state.rooms);
  }
  setChild(event, data) {
    event.persist();
    this.state.rooms.filter((room) => {
      if (room.key === data.key) {
        data.child = event.target.value;
      }
    });
    this.setState({child: event.target.value});
  }
  render() {
    const renderChild = this.state.rooms.map((roomData, index) => {
      return (
        <Col xs={6} lg={3} key={roomData.key}>
          <Panel>
            <Panel.Heading>
              <Panel.Title componentClass="h3">
                {(roomData.key !== 1 ) ? <Checkbox inline onChange={(event) => {this.onRoomSelect(event, roomData)}} checked={roomData.checked}> {roomData.text}</Checkbox> : roomData.text}
              </Panel.Title>
            </Panel.Heading>
            <Panel.Body>
              <Col>
                <FormGroup>
                  <ControlLabel>Adults</ControlLabel>
                  <FormControl componentClass="select" placeholder="select" disabled={!roomData.checked} onChange={(event) => {this.setAdult(event, roomData)}} value={roomData.checked ? roomData.adult : adultsOptions[0].value}>
                    {adults}
                  </FormControl>
                </FormGroup>
              </Col>
              <Col>
              <FormGroup>
                <ControlLabel>Childs</ControlLabel>
                <FormControl componentClass="select" placeholder="select" disabled={!roomData.checked} onChange={(event) => {this.setChild(event, roomData)}} value={roomData.checked ? roomData.child : childrenOptions[0].value}>
                  {children}
                </FormControl>
              </FormGroup>
            </Col>
            </Panel.Body>
          </Panel>
        </Col>
      );
    });
    return (
      <Grid>
        <Row>
          {renderChild}
       </Row>
       <Button onClick={this.submit}>Submit</Button>
      </Grid>
    );
  }
}
function setDataToLocalstorage(roomsData) {
  localStorage.setItem('rooms', JSON.stringify(roomsData));
}
export default App;
