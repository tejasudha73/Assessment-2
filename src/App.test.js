import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import ShallowRenderer from 'react-test-renderer/shallow';

import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders App', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  const result = renderer.getRenderOutput();
  expect(result.type).toBe('div');
});

it('snapshot testinf App', () => {
  const app = shallow(<App />);
  const tree = toJson(app);
  expect(tree).toMatchSnapshot();
});
