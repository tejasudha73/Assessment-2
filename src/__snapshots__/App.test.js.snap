// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App snapshot testing 1`] = `
<div>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        key="1"
        lg={3}
        xs={6}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              Room 1
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Adults
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={false}
                  onChange={[Function]}
                  placeholder="select"
                  value="1"
                >
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Childs
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={false}
                  onChange={[Function]}
                  placeholder="select"
                  value="0"
                >
                  <option
                    key="0"
                    value="0"
                  >
                    0
                  </option>
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        key="2"
        lg={3}
        xs={6}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              <Checkbox
                bsClass="checkbox"
                checked={false}
                disabled={false}
                inline={true}
                onChange={[Function]}
                title=""
              >
                 
                Room 2
              </Checkbox>
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Adults
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="1"
                >
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Childs
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="0"
                >
                  <option
                    key="0"
                    value="0"
                  >
                    0
                  </option>
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        key="3"
        lg={3}
        xs={6}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              <Checkbox
                bsClass="checkbox"
                checked={false}
                disabled={false}
                inline={true}
                onChange={[Function]}
                title=""
              >
                 
                Room 3
              </Checkbox>
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Adults
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="1"
                >
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Childs
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="0"
                >
                  <option
                    key="0"
                    value="0"
                  >
                    0
                  </option>
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        key="4"
        lg={3}
        xs={6}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              <Checkbox
                bsClass="checkbox"
                checked={false}
                disabled={false}
                inline={true}
                onChange={[Function]}
                title=""
              >
                 
                Room 4
              </Checkbox>
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Adults
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="1"
                >
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Childs
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="0"
                >
                  <option
                    key="0"
                    value="0"
                  >
                    0
                  </option>
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Row>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      Submit
    </Button>
  </Grid>
</div>
`;

exports[`snapshot testinf App 1`] = `
<div>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        key="1"
        lg={3}
        xs={6}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              Room 1
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Adults
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={false}
                  onChange={[Function]}
                  placeholder="select"
                  value="1"
                >
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Childs
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={false}
                  onChange={[Function]}
                  placeholder="select"
                  value="0"
                >
                  <option
                    key="0"
                    value="0"
                  >
                    0
                  </option>
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        key="2"
        lg={3}
        xs={6}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              <Checkbox
                bsClass="checkbox"
                checked={false}
                disabled={false}
                inline={true}
                onChange={[Function]}
                title=""
              >
                 
                Room 2
              </Checkbox>
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Adults
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="1"
                >
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Childs
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="0"
                >
                  <option
                    key="0"
                    value="0"
                  >
                    0
                  </option>
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        key="3"
        lg={3}
        xs={6}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              <Checkbox
                bsClass="checkbox"
                checked={false}
                disabled={false}
                inline={true}
                onChange={[Function]}
                title=""
              >
                 
                Room 3
              </Checkbox>
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Adults
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="1"
                >
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Childs
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="0"
                >
                  <option
                    key="0"
                    value="0"
                  >
                    0
                  </option>
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        key="4"
        lg={3}
        xs={6}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              <Checkbox
                bsClass="checkbox"
                checked={false}
                disabled={false}
                inline={true}
                onChange={[Function]}
                title=""
              >
                 
                Room 4
              </Checkbox>
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Adults
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="1"
                >
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Childs
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="select"
                  disabled={true}
                  onChange={[Function]}
                  placeholder="select"
                  value="0"
                >
                  <option
                    key="0"
                    value="0"
                  >
                    0
                  </option>
                  <option
                    key="1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    key="2"
                    value="2"
                  >
                    2
                  </option>
                </FormControl>
              </FormGroup>
            </Col>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Row>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      Submit
    </Button>
  </Grid>
</div>
`;
